tests/test_env_checker.py

```diff
@@ -112,3 +112,58 @@ def step(self, action):
     test_env = TestEnv()
     with pytest.raises(AssertionError, match=error_message):
         check_env(env=test_env)
+
+
+class StepCalledAfterEnvTerminatedException(Exception):
```

Comment:
I don't think that's needed here, as we are not catching it, an assert will be enough.